### PR TITLE
Set correct order for Catalogue Solution filters

### PR DESCRIPTION
**Pull request by kiran-mastek**

### CODE REVIEW

File: src/NHSD.GPIT.BuyingCatalogue.WebApp/Areas/Admin/Models/CatalogueSolutionsModel.cs [hunk 1]
```diff
@@ -1,22 +1,25 @@
...
-            .ToList();
+        private readonly List<PublicationStatus> publicationStatusModels =
+            new[] { 1, 3, 2, 5, 4 }.Select(
+                    s => new PublicationStatus
+                    {
+                        Id = s,
+                        Display = Enums.ToObject<EntityFramework.Models.GPITBuyingCatalogue.PublicationStatus>(s)
+                            .AsString(EnumFormat.DisplayName),
+                    })
+                .ToList();
```
Comment (bjss-chris) [inline src/NHSD.GPIT.BuyingCatalogue.WebApp/Areas/Admin/Models/CatalogueSolutionsModel.cs:18]: Can we use the `Order` property of the `Display` attribute on the `PublicationStatus` and Enums.NET to make the code slightly clearer here? Perhaps something like below?

```c#
public enum PublicationStatus
    {
        [Display(Name = nameof(Draft), Order = 1)]
        Draft = 1,

        [Display(Name = nameof(Unpublished), Order = 3)]
        Unpublished = 2,

        [Display(Name = nameof(Published), Order = 2)]
        Published = 3,

        [Display(Name = nameof(Suspended), Order = 5)]
        Suspended = 4,

        [Display(Name = "In Remediation", Order = 4)]
        InRemediation = 5,
    }
```

```c#
private readonly IReadOnlyList<PublicationStatus> models =
    Enums.GetValues<EntityFramework.Models.GPITBuyingCatalogue.PublicationStatus>(EnumMemberSelection.DisplayOrder)
        .Select(s => new PublicationStatus { Id = (int)s, Display = s.AsString(EnumFormat.DisplayName) })
        .ToList();
```

File: tests/NHSD.GPIT.BuyingCatalogue.WebApp.UnitTests/Areas/Admin/Models/CatalogueSolutionsModelTests.cs [hunk 2]
```diff
@@ -18,12 +15,16 @@ public static void AllPublicationStatuses_StandardCall_ResultAsExpected()
...
 
-            var publicationStatuses = Enum.GetValues<PublicationStatus>();
-            actual.Count.Should().Be(publicationStatuses.Length);
-            foreach (var status in publicationStatuses)
-            {
-                actual.Single(p => p.Id == (int)status).Display.Should().Be(status.AsString(EnumFormat.DisplayName));
-            }
+            actual.Should()
+                .BeEquivalentTo(
+                    new List<NHSD.GPIT.BuyingCatalogue.WebApp.Areas.Admin.Models.PublicationStatus>
```
Comment (bjss-chris) [inline tests/NHSD.GPIT.BuyingCatalogue.WebApp.UnitTests/Areas/Admin/Models/CatalogueSolutionsModelTests.cs:20]: I think the type name can be simplified to `WebApp.Areas.Admin.Models.PublicationStatus` here.

**Review by bjss-chris — CHANGES_REQUESTED** [review]

File: src/NHSD.GPIT.BuyingCatalogue.WebApp/Areas/Admin/Models/CatalogueSolutionsModel.cs (hunk 1 above)
Comment (kiran-mastek) [reply src/NHSD.GPIT.BuyingCatalogue.WebApp/Areas/Admin/Models/CatalogueSolutionsModel.cs]: Good idea, done.

File: tests/NHSD.GPIT.BuyingCatalogue.WebApp.UnitTests/Areas/Admin/Models/CatalogueSolutionsModelTests.cs (hunk 2 above)
Comment (kiran-mastek) [reply tests/NHSD.GPIT.BuyingCatalogue.WebApp.UnitTests/Areas/Admin/Models/CatalogueSolutionsModelTests.cs]: done

**Review by bjss-chris — CHANGES_REQUESTED** [review]

File: src/NHSD.GPIT.BuyingCatalogue.EntityFramework/Models/GPITBuyingCatalogue/PublicationStatus.cs [hunk 3]
```diff
@@ -4,19 +4,18 @@ namespace NHSD.GPIT.BuyingCatalogue.EntityFramework.Models.GPITBuyingCatalogue
...
-        [Display(Name = nameof(Suspended))]
+        [Display(Name = nameof(Suspended), Order = 5)]
         Suspended = 4,
 
-        [Display(Name = "In Remediation")]
+        [Display(Name = "In Remediation", Order = 4)]
         InRemediation = 5,
-    }
-}
+    }}
```
[the remark this answers is not in the document]
Comment (kiran-mastek) [reply src/NHSD.GPIT.BuyingCatalogue.EntityFramework/Models/GPITBuyingCatalogue/PublicationStatus.cs]: done

**Review by bjss-chris — APPROVED** [review]

**Pull request merged**